### ISSUE TITLE
EventBus should propagate deep copies of the published payload

### ISSUE THREAD

**Issue by ggoranov** [dusk-network/dusk-blockchain#292]
Each subscriber should receive deep copy of the message payload to avoid any inconsistency and race conditions.

Below a test demonstrates the issue:

`chain_test.go`
```go
func TestDeepCopy(t *testing.T) {
	eb, _, c := setupChainTest(t, false)
	go c.Listen()

	// Run two subscribers expecting acceptedBlock message
	acceptedBlock1Chan := make(chan message.Message, 1)
	eb.Subscribe(topics.AcceptedBlock, eventbus.NewChanListener(acceptedBlock1Chan))

	acceptedBlock2Chan := make(chan message.Message, 1)
	eb.Subscribe(topics.AcceptedBlock, eventbus.NewChanListener(acceptedBlock2Chan))

	// Propagate accepted block
	var propagatedHeight uint64 = 333
	propagatedBlock := helper.RandomBlock(t, uint64(propagatedHeight), 3)

        // shadow copy here as block.Header is a reference
	msg := message.New(topics.AcceptedBlock, *propagatedBlock)
	eb.Publish(topics.AcceptedBlock, msg)

	// subscriber_1 collecting propagated block
	blkMsg1 := <-acceptedBlock1Chan
	decodedBlk1 := blkMsg1.Payload().(block.Block)

	// subscriber_1 altering the payload
	decodedBlk1.Header.Height = 999

	// subscriber_2 collecting propagated block
	blkMsg2 := <-acceptedBlock2Chan
	decodedBlk2 := blkMsg2.Payload().(block.Block)

	// subscriber_2 ensuring the original height has been received
	assert.True(t, decodedBlk2.Header.Height == propagatedHeight)
}
```

**Comment by ggoranov** [comment]
An option could be to have `payload` of type `IPayload` with `DeepCopy` method (instead of pure interface{})

**Comment by julesdesmit** [comment]
Fixed via #540
